test/e2e/specs/editor/various/post-editor-template-mode.spec.js

```diff
@@ -102,6 +102,35 @@ test.describe( 'Post Editor Template mode', () => {
 		).toBeVisible();
 	} );
 
+	test( 'Allow editing the title of a new custom template', async ( {
+		page,
+		requestUtils,
+		postEditorTemplateMode,
+	} ) => {
+		async function editTemplateTitle( newTitle ) {
+			await page
+				.getByRole( 'button', { name: 'Template Options' } )
+				.click();
+
+			await page
+				.getByRole( 'textbox', { name: 'Title' } )
+				.fill( newTitle );
+
+			const editorContent = page.getByLabel( 'Editor Content' );
+			await editorContent.click();
+		}
+
+		await requestUtils.activateTheme( 'emptytheme' );
+
+		await postEditorTemplateMode.createPostAndSaveDraft();
+		await postEditorTemplateMode.createNewTemplate( 'Foobar' );
+		await editTemplateTitle( 'Barfoo' );
```

Comment:
Oh, I mean to just inline the code without creating an extra function, but this works too, up to you!

Reply:
Ah, cool! But yeah, I think keeping the local function makes it a bit clearer to read, too.